File: core/worker/analysis/infrastructure/adapters/testdata/fixture-java/src/main/java/com/acme/service/UserService.java
package com.acme.service;

import com.acme.model.User;
import com.acme.repo.UserRepository;
import org.springframework.stereotype.Service;

@Service
public class UserService {
    private final UserRepository repository;

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    public User getUser(int id) {
        return repository.find(id);
    }

    public void register(User user) {
        repository.save(user);
    }
}

File: core/worker/analysis/infrastructure/adapters/testdata/fixture-csharp/Services/UserService.cs
using Acme.Data;
using Acme.Models;

namespace Acme.Services;

public class UserService
{
    private readonly UserRepository repository;

    public UserService(UserRepository repository)
    {
        this.repository = repository;
    }

    public User GetUser(int id) => repository.Find(id);

    public void Register(User user) => repository.Save(user);
}
